commands/server.go

```diff
@@ -126,7 +126,13 @@ func server(cmd *cobra.Command, args []string) error {
 	} else {
 		if flagChanged(serverCmd.Flags(), "port") {
 			// port set explicitly by user -- he/she probably meant it!
-			return newSystemErrorF("Port %d already in use", serverPort)
+			switch t := err.(type) {
+			case *net.OpError:
+				err_msg := t.Err.(*os.SyscallError).Err.Error()
+				return newSystemErrorF("Port %d "+err_msg, serverPort)
+			default:
```

Comment:
All of this looks too complicated. What do you get if you just print the original error?